### PR TITLE
Fix #6386

### LINKED ISSUE #6386: window is undefined error in backend functions
We can't check `'Cypress' in window` in back-end.
Let's save whether Cypress is used in _META collection and toggle the setting when starting a e2e command in cli

### PR DESCRIPTION
**Pull request by xmano**
Fixes #6386

### CODE REVIEW

**Comment by xmano** [comment]
Will correct the branch issue and re-submit

File: apps/backend-functions/src/templates/mail.ts [hunk 1]
```diff
@@ -363,7 +363,7 @@ export async function organizationRequestedAccessToApp(org: OrganizationDocument
 }
 
 export async function userFirstConnexion(user: PublicUser): Promise<EmailRequest> {
-  const supportEmail = ('Cypress' in window) ? testEmail : getSupportEmail(user._meta.createdFrom);
+  const supportEmail = (e2e_mode) ? testEmail : getSupportEmail(user._meta.createdFrom);
```
Comment (RemcoSimonides) [inline apps/backend-functions/src/templates/mail.ts:366]: No need for `()` around e2e_mode

**Review by RemcoSimonides — COMMENTED** [review]
What happened with the idea of setting a variable when starting an e2e process instead of defining it statically?

File: apps/backend-functions/src/templates/mail.ts (hunk 1 above)
Comment (xmano) [reply apps/backend-functions/src/templates/mail.ts:366]: Ok. will take care of this along with other review comments if any.

**Comment by phpgeekfr** [comment]
@xmano I cleaned branch history so this should be clean. But ask me again before merging please

File: apps/backend-functions/src/templates/mail.ts (hunk 1 above)
Comment (xmano) [reply apps/backend-functions/src/templates/mail.ts:366]: @george43g : plesae let me know if using env config is ok or we want more changes like setting a parameter during e2e config.

**Comment by xmano** [comment]
> @xmano I cleaned branch history so this should be clean. But ask me again before merging please

Sure!

**Comment by george43g** [comment]
hey mate,

apps/backend-functions/src/environments/environment.e2e.ts

create the above file and have it as a build config. 
so

nx build backend-functions --configuration=e2e ...(or =emulator, depending on the real purpose of these changes)

should select that build config (using that file with a fileReplacement) against any projectId

**Comment by xmano** [comment]
I made changes to backend-functions environment. However due to some pre-commit issue I cannot commit.

Here are the main diffs:

```
diff --git a/angular.json b/angular.json
index e57baed21..0e88ec694 100644
--- a/angular.json
+++ b/angular.json
@@ -208,7 +208,7 @@
                 },
                 {
                   "replace": "apps/backend-functions/src/environments/environment.ts",
-                  "with": "apps/backend-functions/src/environments/environment.prod.ts"
+                  "with": "apps/backend-functions/src/environments/environment.ci.ts"
                 }
               ]
             }
```

backend-function environment

```
diff --git a/apps/backend-functions/src/environments/environment.prod.ts b/apps/backend-functions/src/environments/environment.prod.ts
index 20ab18f51..ede1224c6 100644
--- a/apps/backend-functions/src/environments/environment.prod.ts
+++ b/apps/backend-functions/src/environments/environment.prod.ts
@@ -6,6 +6,8 @@
  */
 import * as functions from 'firebase-functions';

+export const e2e_mode = false;
+
 export {
   factoryContract,
   backupBucket,
```

**Comment by xmano** [comment]
@george43g : kindly review the diffs posted as comment as above. I would like to work on this and close it.

**Comment by george43g** [comment]
you have made the default mode e2e mode. We still havent fully transitioned to this... 
also, you have not referenced your new .ci file in angular.json. This mode is to be used during e2e testing correct?

**Pull request merged**